apps/test/solutions/applab/ec_design.js

```diff
@@ -593,6 +591,79 @@ module.exports = {
     },
 
     {
+      description: "exercise duplicate button on elements",
+      editCode: true,
+      xml: '',
+      runBeforeClick: function (assert) {
+
+        // Switch to design mode
+        var designModeButton = $('#designModeButton');
+        designModeButton.click();
+
+        // Add an image
+        testUtils.dragToVisualization('IMAGE', 0, 0);
+        var designModeViz = $('#designModeViz');
+        var newImage = designModeViz.find('img');
+        assert.equal(newImage.length, 1);
+
+        // Duplicate the image
+        var imageDuplicateButton = $('#designWorkspaceBody').find('button:contains(Duplicate)')[0];
+        ReactTestUtils.Simulate.click(imageDuplicateButton);
+        assert.equal(designModeViz.find('img').length, 2);
+
+        // Add a chart
+        testUtils.dragToVisualization('CHART', 0, 0);
+        var newChart = designModeViz.find('.chart');
+        assert.equal(newChart.length, 1);
+
+        // Duplicate the chart
+        var chartDuplicateButton = $('#designWorkspaceBody').find('button:contains(Duplicate)')[0];
+        ReactTestUtils.Simulate.click(chartDuplicateButton);
+        assert.equal(designModeViz.find('.chart').length, 2);
+
+        Applab.onPuzzleComplete();
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      }
+    },
+
+    {
+      description: "exercise copy paste button on elements",
+      editCode: true,
+      xml: '',
+      runBeforeClick: function (assert) {
+
+        // Switch to design mode
+        var designModeButton = $('#designModeButton');
+        designModeButton.click();
+
+        // Add an image
+        testUtils.dragToVisualization('IMAGE', 0, 0);
+        var designModeViz = $('#designModeViz');
+        var newImage = designModeViz.find('img');
+        assert.equal(newImage.length, 1);
+
+        // Copy and paste the image
+        var copy = testUtils.createKeyEvent('keydown', 67, true);
+        var paste = testUtils.createKeyEvent('keydown', 86, true);
+
+        var designModeElement = document.getElementById('designModeViz');
+        designModeElement.dispatchEvent(copy);
+        designModeElement.dispatchEvent(paste);
+
+        assert.equal(designModeViz.find('img').length, 2);
```

Comment:
If possible, it would be great to paste a second time and verify the final position of the final element, which I would expect to be (20, 20)